core/example/solvers/particles.cpp

```diff
@@ -0,0 +1,73 @@
+#include "definitions.h"
+#include "particles.h"
+
+// function to initialize particles as a NaCl crystal of length crystal_size
+void initializeParticles(ParticleList particles, int crystal_size)
```

Comment:
Will the particle list in this case always be on the host or use Cuda UVM memory? If not you need to place this loop in a parallel for